book/style_guide.md

```diff
@@ -0,0 +1,221 @@
+# Best practices
+
+This page lists our best practices we advice to follow. Keep in mind that they are not required to be
+used in external repositories (not ours), you can change them in the way you want,
+but please be consistent and follow your rules.
+
+## Formatting
+
+### Defaults
+
+**It's recommended to** assume that by default no spaces or tabs allowed, but the following rules define where they are allowed.
+
+### Basic
+
+- **It's recommended to** put one space before and after pipe `|` symbol, commands, subcommands, their options and arguments.
+- **It's recommended to** omit commas between list items.
+
+Correct:
+
+```nushell
+'Hello, Nushell! This is a gradient.' | ansi gradient --fgstart '0x40c9ff' --fgend '0xe81cff'
+```
+
+Incorrect:
+
+```nushell
+'Hello, Nushell! This is a gradient.' |  ansi gradient --fgstart '0x40c9ff' --fgend '0xe81cff' # two many spaces after "|": 2 instead of 1
+```
+
+#### One-line format
+
+One-line format is a format for writing all commands in one line.
+
+1. parameters:
+   1. **It's recommended to** put no spaces before and after pipe `|` symbol denoting block or closure parameter list beginning.
+   2. **It's recommended to** put one space after comma `,` after block or closure parameter parameter.
```

Comment:
too many parameters here :wink:

Reply:
Do you suggest splitting these rules two several ones?

Reply:
ooh no no, sorry.
i meant you wrote "parameter parameter" :yum:

Reply:
Fixed 🚀

Reply:
my bad for not being clear :laughing: 
thanks emily :yum: